src/screens/FeedScreen/RenderList.js

```diff
@@ -112,10 +114,54 @@ const RenderListFeed = (props) => {
 
   const hasComment = getCommentLength(item.latest_reactions.comment) > 0;
 
-  const contentLinkHeight = () => {
-    const haveLength = hasComment ? getHeightReaction() / 2.2 : getHeightReaction() / 1.6;
-    return dimen.size.FEED_CURRENT_ITEM_HEIGHT - getHeightHeader() - getHeightFooter() - haveLength;
+  const isBlurred = item?.isBlurredPost && item?.anonimity;
+
+  const renderWriteComment = () => {
+    return (
+      !isBlurred && (
+        <TouchableOpacity
+          testID="writeComment"
+          onPress={() => onPressComment(isHaveSeeMore)}
+          style={styles.contentReaction(getHeightReaction())}>
+          <WriteComment
+            postId={''}
+            username={handleUserName(item)}
+            value={''}
+            onChangeText={() => {}}
+            onPress={() => {}}
+            loadingPost={false}
+            isViewOnly={true}
+            withAnonymityLabel={false}
+          />
+        </TouchableOpacity>
+      )
+    );
+  };
+
+  const commentHeight = () => {
+    if (isBlurred && !hasComment) {
+      return 0;
+    }
+    const isSingleComment = getTotalReaction(item) === 1;
+    const commentSectionHeight = getHeightReaction() - getHeightFooter();
+    return isSingleComment ? commentSectionHeight - 20 : commentSectionHeight;
+  };
+
+  const {onLayoutTopicChip} = useCalculationContent();
+  const calculateLineTopicChip = (nativeEvent) => {
+    onLayoutTopicChip(nativeEvent, 1);
   };
+
+  const topicBottomPosition = () => {
+    if (hasComment) {
+      if (getTotalReaction(item) === 1) {
+        return getHeightReaction() - 20;
+      }
+      return getHeightReaction();
+    }
+    return getHeightFooter();
+  };
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for rendering comments and calculating their height seems complex. Consider simplifying it or breaking it down into smaller, more manageable functions. This will make the code easier to understand and maintain. Also, ensure that all the functions like `getHeightReaction()`, `getHeightFooter()`, `getTotalReaction(item)` are properly handling potential null or undefined values to avoid runtime errors.



<!-- This is an auto-generated comment by OSS CodeRabbit -->